de-lib/src/main/java/org/iplantc/de/fileViewers/client/callbacks/ShareAnonymousCallback.java

```diff
@@ -60,32 +60,33 @@ public void onFailure(Integer statusCode, Throwable caught) {
     }
 
     @Override
-    public void onSuccess(String result) {
+    public void onSuccess(List<String> urls) {
         if (container != null) {
             container.unmask();
         }
-        JSONObject obj = jsonUtil.getObject(result);
-        JSONObject paths = jsonUtil.getObject(obj, "paths");
-        showShareLink(jsonUtil.getString(paths, file.getPath()));
+
+        showShareLink(urls.stream()
+                          .filter(path -> !diskResourceUtil.isGenomeIndexFile(path))
+                          .collect(Collectors.toList()));
```

Comment:
The `shareWithAnonymous` call now uses a callback converter to pass this callback a list of sorted URL strings for all the paths shared with the anon user.

So now this callback needs to filter out the genome index files from the dialog displayed on success, since the user can't use the URLs to those index files in the genome browser sites (but the index files are still required to be readable by those sites).